src/mdns/mdns.hpp

```diff
@@ -143,9 +143,10 @@ class Publisher : private NonCopyable
      */
     struct DiscoveredHostInfo
     {
-        std::string mHostName;  ///< Full host name.
-        AddressList mAddresses; ///< IP6 addresses.
-        uint32_t    mTtl = 0;   ///< Host TTL.
+        std::string mHostName;       ///< Full host name.
+        AddressList mAddresses;      ///< IP6 addresses.
+        uint32_t    mNetifIndex = 0; ///< Network interface.
```

Comment:
It seems POSIX defines network interface index to `unsigned int`. Should we use that type instead?

https://pubs.opengroup.org/onlinepubs/9699919799/functions/if_nametoindex.html

Reply:
I am not sure. `mDNSResponder` uses `uint32_t` in its API, and `Avahi` has `typedef int AvahiIfIndex`.

Independent of whether we want to change this type or not, I would suggest such a change to be done in its own PR and change all existing definitions of `mNetifIndex` in `Publisher` module. This PR is adding the missing in `mNetIfIdex` in `DiscoveredHostInfo` to make it similar to `DiscoveredServiceInfo`.